### PR TITLE
Add support for true color in Terminal output.

### PR DESCRIPTION
**Pull request by grynspan**
Ubuntu and Windows both support true color output in their respective Terminal applications, so we can support it as well. (It's admittedly a subtle change from 256-color support.)

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

File: Sources/Testing/Events/Recorder/Event.ConsoleOutputRecorder.swift [hunk 1]
```diff
@@ -148,17 +148,19 @@ extension Event.Symbol {
   /// - Returns: A string representation of `self` appropriate for writing to
   ///   a stream.
   fileprivate func stringValue(options: Set<Event.ConsoleOutputRecorder.Option>) -> String {
+    var useColorANSIEscapeCodes = false
```
Comment (SeanROlszewski) [inline Sources/Testing/Events/Recorder/Event.ConsoleOutputRecorder.swift:151]: If you wanted to you could use an if expression here and avoid making this mutable. 

```swift
let useColorANSIEscapeCodes = if let colorBitDepth = options.colorBitDepth {
  colorBitDepth >= 4
} else {
  false
}

**Review by SeanROlszewski — APPROVED** [review]

File: Sources/Testing/Events/Recorder/Event.ConsoleOutputRecorder.swift (hunk 1 above)
Comment (grynspan) [reply Sources/Testing/Events/Recorder/Event.ConsoleOutputRecorder.swift:151]: Sure, but this is one line shorter. :D

**Pull request merged**